pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -39,42 +48,89 @@ func (*AgentClusterInstall) Dependencies() []asset.Asset {
-	// }
-
-	// var numberOfWorkers int = 0
-	// for _, compute := range installConfig.Config.Compute {
-	// 	numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
-	// }
-	// agentClusterInstall.Spec.ProvisionRequirements.WorkerAgents = numberOfWorkers
-	// a.Config = agentClusterInstall
-
-	// agentClusterInstallData, err := yaml.Marshal(agentClusterInstall)
-	// if err != nil {
-	// 	return errors.Wrap(err, "failed to marshal agent installer AgentClusterInstall")
-	// }
-
-	// a.File = &asset.File{
-	// 	Filename: agentClusterInstallFilename,
-	// 	Data:     agentClusterInstallData,
-	// }
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
 
+	if installConfig.Config != nil {
+		var numberOfWorkers int = 0
+		for _, compute := range installConfig.Config.Compute {
+			numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
+		}
+
+		clusterNetwork := []hiveext.ClusterNetworkEntry{}
+		for _, cn := range installConfig.Config.Networking.ClusterNetwork {
+			_, cidr, err := net.ParseCIDR(cn.CIDR.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to parse ClusterNetwork CIDR")
+			}
+			err = validate.SubnetCIDR(cidr)
+			if err != nil {
+				return errors.Wrap(err, "failed to validate ClusterNetwork CIDR")
+			}
+
+			entry := hiveext.ClusterNetworkEntry{
+				CIDR:       cidr.String(),
+				HostPrefix: cn.HostPrefix,
+			}
+			clusterNetwork = append(clusterNetwork, entry)
+		}
+
+		serviceNetwork := []string{}
+		for _, sn := range installConfig.Config.Networking.ServiceNetwork {
+			cidr, err := ipnet.ParseCIDR(sn.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to parse ServiceNetwork CIDR")
+			}
+			serviceNetwork = append(serviceNetwork, cidr.String())
+		}
+
+		agentClusterInstall := &hiveext.AgentClusterInstall{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      getAgentClusterInstallName(installConfig),
+				Namespace: getObjectMetaNamespace(installConfig),
+			},
+			Spec: hiveext.AgentClusterInstallSpec{
+				ImageSetRef: &hivev1.ClusterImageSetReference{
+					Name: "openshift-4.11",
+				},
+				ClusterDeploymentRef: corev1.LocalObjectReference{
+					Name: getClusterDeploymentName(installConfig),
+				},
+				Networking: hiveext.Networking{
+					ClusterNetwork: clusterNetwork,
+					ServiceNetwork: serviceNetwork,
+				},
+				SSHPublicKey: strings.Trim(installConfig.Config.SSHKey, "|\n\t"),
+				ProvisionRequirements: hiveext.ProvisionRequirements{
+					ControlPlaneAgents: int(*installConfig.Config.ControlPlane.Replicas),
+					WorkerAgents:       numberOfWorkers,
+				},
+			},
+		}
+
+		// set APIVIP and IngressVIP only for non SNO cluster with Baremetal and Vsphere platforms
+		// SNO cluster is determined by number of ControlPlaneAgents which should be 1
+		if (int(*installConfig.Config.ControlPlane.Replicas) > 1) &&
+			(installConfig.Config.Platform.Name() == baremetaltypes.Name ||
+				installConfig.Config.Platform.Name() == vspheretypes.Name) {
```

Comment:
Isn't the check of the platform type effectively handled in `getVIPs()`?

Reply:
yes, fixed it.